### ISSUE TITLE
Improve error message for invalid recipient in conversations

### ISSUE THREAD

**Issue by goobertron** [diaspora/diaspora#7800]
At the moment, if you enter the name of someone with whom you don't have mutual sharing, and therefore with whom you can't start a conversation, the error message reads:

> Hey, you need to add a recipient first!

I think this alert should only be shown if the recipient box is blank. If there is something in the box, the alert should be something more like:

> You can’t start a conversation with this person

If someone wants to get really clever with this, they could code it so that this alert is shown if the name/ID entered is a valid diaspora account but of someone with whom the sender does not have mutual sharing, and show a different alert if the string entered does not match any known name/ID, something like:

> I can’t find anyone matching this name

But that might be needlessly complex.

**Comment by rtrunkhill** [comment]
Instead of changing the code, would combining the two messages work?

ie. "Hey, you either cant start a conversation with this person or you forgot to add a recipient"

**Comment by DrTao** [comment]
I second this, and even the combined message would provide far more helpful context than the current message.

**Comment by HankG** [comment]
I agree the UI is a bit confusing to the uninitiated.  The problem is that the "To:" fields are actually bubbles not the search field for adding a new recipient.  So like in this attached screenshot there is one recipient and a failed search for a second. In fact if you paste in the name and just leave that as is, or tab off of it without selecting and don't add the card to the actual "to" list it gives the same error.  A suggestion I was thinking is that we label the text box to be something like "Add Recepient" (or something).  We may want to work on tweaking the error messages too but they are technically consistent with the behavior :
![image](https://user-images.githubusercontent.com/1683691/46587283-f3707880-ca57-11e8-9209-dea9bf396491.png)

**Comment by HankG** [comment]
While discussing this is there any interest to add the usernames in the recipient list when viewing a conversation?  I'm torn on this personally since almost none of my contacts have generic icons but I ran across this case in my testing on my dev server where no one has icons and thought I'd bring it up.  If there is I could add a new issue number and start looking into it for real (this is something quick and dirty): 
![image](https://user-images.githubusercontent.com/1683691/46587645-47319080-ca5d-11e8-8bbf-527a5d5ee5fc.png)

**Comment by jhass** [comment]
I think the empty "To" section is a bit confusing with that approach, can we solve this with a better placeholder like "Search contacts" and/or dynamically changing "To" to read something like "Add recipient" while the recipient list is empty?

I think just icons in the participant list is fine.

**Comment by HankG** [comment]
@jhass yes I originally was trying to do something quick and dirty and have it read in light italics "No Recipients" or something like that but couldn't figure it out for the screenshot.  The composition is in the HAML but the heavy lifting of the work is being done in Backbone IIRC.  I have to think between the two we should be able to do something like what you are describing.

**Comment by goobertron** [comment]
@HankG: 'is there any interest to add the usernames in the recipient list when viewing a conversation?'

@jhass: 'I think just icons in the participant list is fine.'

I agree with Jonne. The user (screen) name is available as a tooltip on hover in the desktop version, and that's enough. And it's a lot better UI – even in the example with three names next to the icons, it looks very cumbersome.
